lib/log_parser.rb

```diff
@@ -1,14 +1,28 @@
+require "json"
 class LogParser
-  def initialize(file_name)
-    @file_name = file_name
-    raise "File not found" unless File.exists?(@file_name)
+  def initialize(file_path)
+    @file_path = file_path
+    raise "File not found" unless File.exists?(@file_path)
+    @file = File.open(@file_path)
+    @file_data = @file.readlines.map(&:chomp)
   end
 
   def get_first_line
-    file = File.open(@file_name)
-    file_data = file.readlines.map(&:chomp)
-    first_line = file_data[0]
-    file.close
-    first_line
+    first_line = @file_data[0]
+  end
+
+  def get_log_data
+    @file.close
+    data = {:lines => self.file_length}
+    '"%{file_name}": %{json}' % {file_name: self.file_name, json:JSON.pretty_generate(data)}
```

Comment:
```suggestion
    '"%{file_name}": %{json}' % {file_name: file_name, json:JSON.pretty_generate(data)}
```

Não precisa do self aqui pois o ruby sabe que é um método interno da classe